packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -72,14 +73,18 @@ class MojoAssetBundle extends AssetBundle {
     _imageCache = null;
   }
 
-  Future<ui.Image> _fetchImage(String key) async {
-    return await decodeImageFromDataPipe(await load(key));
+  Future<ui.Image> _fetchImage(String key, [ AssetResolver resolver ]) async {
+    if (resolver != null)
+      key = await resolver.resolve(key);
+    if (_imageCache.containsKey(key))
+      return _imageCache[key];
+    ui.Image value = await decodeImageFromDataPipe(await load(key));
+    _imageCache[key] = value;
+    return value;
```

Comment:
I think there is something wrong with this rewrite as a few times I got blank space where an image should have loaded. I didn't dig into it too far because I wanted to put this up for feedback; in particular, if the asset resolver is persisted in the asset bundle somehow, then this logic would be restructured anyway.